pkg/upgrade/base.go

```diff
@@ -17,12 +17,34 @@ import (
 	clusterapiv1alpha1client "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
 )
 
+type ConfigMapNamespacer interface {
```

Comment:
The thing I really like about this is that our dependencies are listed out very clearly here. I know this app only needs to Get and Create config maps, List nodes, Get and List pods. It does not need watches or anything fancy.

If these change that means there is change in functionality (probably) and I'd fully expect the tests to have to change to reflect that.